src/ansiblelint/config.py

```diff
@@ -164,5 +165,27 @@ def ansible_version(version: str = "") -> Version:
     return Version(version)
 
 
+def path_inject() -> None:
+    """Add python interpreter path to top of PATH to fix outside venv calling."""
+    # This make it possible to call ansible-lint that was installed inside a
+    # virtualenv without having to pre-activate it. Otherwise subprocess will
+    # either fail to find ansible executables or call the wrong ones.
+    #
+    # This must be run before we do run any subprocesses, and loading config
+    # does this as part of the ansible detection.
+    paths = [x for x in os.environ.get("PATH", "").split(os.pathsep) if x]
+    ansible_path = shutil.which("ansible")
+    if ansible_path:
+        ansible_path = os.path.dirname(ansible_path)
+    py_path = os.path.dirname(sys.executable)
+    # Determine if we need to manipulate PATH
+    if ansible_path not in paths or py_path != ansible_path:  # pragma: no cover
+        # tested by test_call_from_outside_venv but coverage cannot detect it
+        paths.insert(0, py_path)
+        os.environ["PATH"] = os.pathsep.join(paths)
+        print(f"WARNING: PATH altered to include {py_path}", file=sys.stderr)
```

Comment:
Could you use a warning subsystem instead?
```suggestion
        warnings.warn(f"PATH altered to include {py_path}", UserWarning, stacklevel=2)
```

Reply:
Per https://docs.python.org/3/library/warnings.html:
> Warning messages are normally written to [sys.stderr](https://docs.python.org/3/library/sys.html#sys.stderr), but their disposition can be changed flexibly, from ignoring all warnings to turning them into exceptions.

and `warnings.showwarning()`:
> Write a warning to a file. The default implementation calls `formatwarning(message, category, filename, lineno, line)` and writes the resulting string to file, which defaults to [sys.stderr](https://docs.python.org/3/library/sys.html#sys.stderr). You may replace this function with any callable by assigning to `warnings.showwarning`.

Reply:
Use of warnings module is outside the scope.